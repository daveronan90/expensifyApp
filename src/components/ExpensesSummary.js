import numeral from "numeral";
import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import expensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal / 100).format(
    "$0,00.00"
  );

  return (
    <div>
      <h1>
        Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = ({ expenses, filters }) => {
  const visibileExpenses = selectExpenses(expenses, filters);
  return {
    expensesCount: expenses ? 0 : visibileExpenses.length,
    expensesTotal: expensesTotal(visibileExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
