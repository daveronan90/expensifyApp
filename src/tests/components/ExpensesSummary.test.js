import { shallow } from "enzyme";
import React from "react";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary />);

  expect(wrapper).toMatchSnapshot();
});

test("Should render Expenses Summary with 2 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expensesTotal={100} expensesCount={2} />
  );

  expect(wrapper).toMatchSnapshot();
});
