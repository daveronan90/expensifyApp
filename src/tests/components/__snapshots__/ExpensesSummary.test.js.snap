// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenses Summary with 2 expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $1.00
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.50
  </h1>
</div>
`;
